framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -374,8 +437,47 @@ public static boolean canInsertJavaExpression(JavaExpression expr) {
      * new and old value are taken (according to the lattice). Note that this happens per hierarchy,
      * and if the store already contains information about a hierarchy for which {@code value} does
      * not contain information, then that information is preserved.
+     *
+     * <p>This method does nothing if {@code expr} cannot be inserted into the store.
```

Comment:
The Javadoc doesn't say that this method will not add expressions that are nondeterministic.  Nondeterministic expression can be inserted in the store, so it's not enough to say `This method does nothing if {@code expr} cannot be inserted into the store.`.